homeassistant/components/tcp/common.py

```diff
@@ -0,0 +1,173 @@
+"""Common code for TCP component."""
+from __future__ import annotations
+
+import logging
+import select
+import socket
+import ssl
+from typing import Final
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import (
+    PLATFORM_SCHEMA as PARENT_PLATFORM_SCHEMA,
+    SensorEntity,
+)
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PAYLOAD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_TIMEOUT,
+    CONF_UNIT_OF_MEASUREMENT,
+    CONF_VALUE_TEMPLATE,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import TemplateError
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.template import Template
+from homeassistant.helpers.typing import ConfigType, StateType
+
+from .const import (
+    CONF_BUFFER_SIZE,
+    CONF_VALUE_ON,
+    DEFAULT_BUFFER_SIZE,
+    DEFAULT_NAME,
+    DEFAULT_SSL,
+    DEFAULT_TIMEOUT,
+    DEFAULT_VERIFY_SSL,
+)
+from .model import TcpSensorConfig
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+TCP_PLATFORM_SCHEMA: Final = PARENT_PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PORT): cv.port,
+        vol.Required(CONF_PAYLOAD): cv.string,
+        vol.Optional(CONF_BUFFER_SIZE, default=DEFAULT_BUFFER_SIZE): cv.positive_int,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+        vol.Optional(CONF_UNIT_OF_MEASUREMENT): cv.string,
+        vol.Optional(CONF_VALUE_ON): cv.string,
+        vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+        vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+        vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean,
+    }
+)
+
+
+class TcpSensor(SensorEntity):
```

Comment:
I think binary sensor platform shouldn't inherit `SensorEntity`.
```suggestion
class TcpSensor(Entity):
```